ipywidgets/widgets/widget.py

```diff
@@ -317,6 +318,47 @@ def _call_widget_constructed(widget):
         if Widget._widget_construction_callback is not None and callable(Widget._widget_construction_callback):
             Widget._widget_construction_callback(widget)
 
+    @classmethod
+    def handle_control_comm_opened(cls, comm, msg):
+        version = msg.get('metadata', {}).get('version', '')
+        if version.split('.')[0] != CONTROL_PROTOCOL_VERSION_MAJOR:
+            raise ValueError("Incompatible widget control protocol versions: received version %r, expected version %r"%(version, __control_protocol_version__))
+
+        cls._control_comm = comm
+        cls._control_comm.on_msg(cls.handle_control_comm_msg)
+
+    @classmethod
+    def handle_control_comm_msg(cls, msg):
+        # This shouldn't happen unless someone calls this method manually
+        if cls._control_comm is None:
+            raise RuntimeError('Control comm has not been properly opened')
+
+        data = msg['content']['data']
+        method = data['method']
+
+        if method == 'request_states':
+            # Send back the full widgets state
+            cls.get_manager_state()
+            widgets = Widget.widgets.values()
```

Comment:
I see that the other methods here are static (e.g. `handle_comm_opened`), and then explicitly use `Widget.` to access other static / class attributes. I kind of agree that it makes more sense as a classmethod, but just noticed this as a difference. I'm just curious why there is `Widget.widgets` here, instead of `cls.widgets`?

Reply:
Good catch! I'll fix it, thanks